### PR TITLE
Add Contact Flow Refactor

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Ref: https://github.com/foam-framework/foam2/pull/3298

- Add IconTextFieldView
Screen Shot 2020-03-18 at 12 28 03 PM

- Add subtitle to Section
Screen Shot 2020-03-12 at 2 22 19 PM

### CODE REVIEW

File: src/foam/u2/detail/SectionView.js [hunk 1]
```diff
@@ -53,13 +53,21 @@ foam.CLASS({
...
+        .add(self.slot(function(section, showTitle, section$title, section$subTitle) {
           if ( ! section ) return;
           return self.Rows.create()
             .show(section.createIsAvailableFor(self.data$))
             .callIf(showTitle && section$title, function() {
               this.start('h2').add(section$title).end();
             })
+            .callIf(section$subTitle, function() {
+              this.start().style({
+                'color': '#8e9090',
```
Comment (jlhughes) [inline src/foam/u2/detail/SectionView.js]: Can you check for other colour assignment, I believe it's /* NAME */, fallback   or something to this affect.  With colours taken from theme.

File: src/foam/u2/view/IconTextFieldView.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
```
Comment (jlhughes) [inline src/foam/u2/view/IconTextFieldView.js:2]: Update this header to the more recent.   and 2020.

File: src/foam/u2/detail/SectionView.js (hunk 1 above)
Comment (zahnno) [reply src/foam/u2/detail/SectionView.js]: Macro reference example `color: /*%BLACK%*/ #1e1f21` I think you'll be looking for  `/*%GREY4%*/` for the subtitle.

File: src/foam/u2/detail/SectionView.js [hunk 3]
```diff
@@ -53,13 +53,21 @@ foam.CLASS({
...
+        .add(self.slot(function(section, showTitle, section$title, section$subTitle) {
           if ( ! section ) return;
           return self.Rows.create()
             .show(section.createIsAvailableFor(self.data$))
             .callIf(showTitle && section$title, function() {
               this.start('h2').add(section$title).end();
             })
+            .callIf(section$subTitle, function() {
+              this.start().style({
+                'color': '/*%GREY4%*/ #8e9090',
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionView.js]: Does variable substitution work when you inline it? If not, better to add to css: block.

File: src/foam/u2/view/IconTextFieldView.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'IconTextFieldView',
+  extends: 'foam.u2.View',
+
+  requires: [
+    'foam.u2.TextField'
+  ],
+
```
Comment (kgrgreer) [inline src/foam/u2/view/IconTextFieldView.js:17]: Add documentation:

File: src/foam/u2/view/IconTextFieldView.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      name: 'type'
+    },
+    {
+      class: 'String',
+      name: 'placeHolder'
+    }
+  ],
+
+  methods: [
+    function initE() {
```
Comment (kgrgreer) [inline src/foam/u2/view/IconTextFieldView.js:42]: Should have a fromProperty() method similar to foam.u2.TextField's.

**Pull request merged**